File: objc/Demo Objc/Controller/DownloadTableViewController.m
//
//  DownloadTableViewController.m
//  Demo Objc
//
//  Created by Eduardo Mello de Macedo | Stone on 03/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

#import "DownloadTableViewController.h"
#import "NSString+Utils.h"

@interface DownloadTableViewController ()

@property (strong, nonatomic) IBOutlet UIButton *downloadButton;

@property (weak, nonatomic) IBOutlet UILabel *feedback;

@end

@implementation DownloadTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = [kTitleTableDownload localize];
    [self.downloadButton setTitle:[kButtonDownload localize] forState:UIControlStateNormal];
    
    self.overlayView = [[UIView alloc] initWithFrame:[UIScreen mainScreen].bounds];
    self.overlayView.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.5];
    self.activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
    self.activityIndicator.center = self.overlayView.center;

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)performDownload:(id)sender {
    NSLog(@"Deprecated feature");
}

@end

File: swift/Demo Swift/DownloadTableViewController.swift
//
//  DownloadTableViewController.swift
//  Demo Swift
//
//  Created by Eduardo Mello de Macedo | Stone on 15/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

import UIKit
import StoneSDK

class DownloadTableViewController: UIViewController {

    @IBOutlet weak var feedbackLabel: UILabel!
    
    var loadingView: LoadingView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = "Download de Tabelas"
        
        self.loadingView = LoadingView.init(frame: UIScreen.main.bounds)
        self.navigationController?.view.addSubview(self.loadingView!)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    @IBAction func performDownload(_ sender: Any) {
        /*
            Efetuando o download das tabelas
         */
        feedbackLabel.text = "Carregando..."
        self.loadingView.show()
        
        STNTableDownloaderProvider.downLoadTables { (succeeded, error) in
            DispatchQueue.main.async {
                if succeeded
                {
                    self.feedbackLabel.text = "Download Realizado"
                }
                else
                {
                    self.feedbackLabel.text = error.debugDescription
                }
                self.loadingView.hide()
            }
        }
    }
}
